common/archiver/interface.go

```diff
@@ -78,8 +78,19 @@ type (
 
 	// HistoryArchiver is used to archive history and read archived history
 	HistoryArchiver interface {
-		Archive(context.Context, URI, *ArchiveHistoryRequest, ...ArchiveOption) error
+        // Archive is used to archive a Workflow's history. When the context expires the method should stop trying to archive.
```

Comment:
nit: Thoughts on tense of `Archive is used to archive ...` vs `Archive will archive ...`

Reply:
🤔  Well, since this is in Interface, and not actual method function - its not like these will actually DO something.
So, to say "Archive will archive" is a bit misleading... "Archive is used to" is more accurate as it is telling the developer what the interface method is meant to do in their own implementation.
I see your point, but in this case I think this is fine. TBH I didn't screen this rhetoric too much - mostly copied it from the README verbatim with small changes - so there is some room to improve.